src/env_paths/mod.rs

```diff
@@ -163,27 +188,87 @@ pub fn javac() -> Option<PathBuf> {
     )
 }
 
-/// Returns the JAVA_HOME path by attempting to discover it.
+/// Returns the `JAVA_HOME` path by attempting to discover it.
 /// 
 /// First, if the `$JAVA_HOME` environment variable is set and points to a directory that exists,
 /// that path is returned.
 /// Otherwise, a series of common installation locations is used,
 /// based on the current platform (macOS, Linux, Windows).
 pub fn java_home() -> Option<PathBuf> {
-    env::var(JAVA_HOME).ok()
+    env_var(JAVA_HOME).ok()
         .and_then(PathExt::path_if_exists)
         .map(PathBuf::from)
         .or_else(find_java_home)
 }
 
-/// Returns the source version for compilation
-/// from `JAVA_SOURCE_VERSION`,
-pub fn java_source_version() -> Option<String> {
-    env::var(JAVA_SOURCE_VERSION).ok()
+/// Returns the source version for compilation from environment variable `JAVA_SOURCE_VERSION`.
+pub fn java_source_version() -> Option<i32> {
+    env_var(JAVA_SOURCE_VERSION).ok()?.parse().ok()
+}
+
+/// Returns the target version for compilation from environment variable `JAVA_TARGET_VERSION`.
+pub fn java_target_version() -> Option<i32> {
+    env_var(JAVA_TARGET_VERSION).ok()?.parse().ok()
+}
+
+/// Returns the major version number of the `javac` compiler.
+pub fn check_javac_version(java_home: impl AsRef<Path>) -> std::io::Result<i32> {
```

Comment:
Question: is there a particular reason why Java version integers are represented by `i32` and not an unsigned integer type? Is it possible to have a negative version number in some cases?

(Seems like you already check for a negative result from the std::process::Command and convert it into an error, so I'd assume all valid version numbers would be positive.)